### PR TITLE
Fix upgrade.

### PR DESCRIPTION
**Pull request by costinm**
Disable backward-incompatible mixer config.

### CODE REVIEW

**Comment by costinm** [comment]
Re 'close': not clear why it is needed (not controlled by anything, not use case to change it), and not clear if it was tested for upgrade. Want to reduce risks.

**Comment by costinm** [comment]
I mean: the mixer client should fail close for policy and open for telemetry, by default.

If we find a need to change the default, and define an API for this - we can start using the mixer plugin in pilot to generate the attribute. Until this happens - there is not point in pilot generating redundant config, 
and no point in mixerclient having a different default. So this change is not needed, and won't be needed until the API is defined.

**Comment by kyessenov** [comment]
Per @liminw we need FAIL_CLOSE for a usable RBAC. This is not a new field, so it should not cause problems.

**Comment by kyessenov** [comment]
Please remove FAIL_CLOSE conditioned on a variable. The rest looks fine.

**Comment by costinm** [comment]
Added back for now the fail close - it is not the right way to enable RBAC, please fix it in mixerclient code and remove it from pilot. Policy and RBAC can't fail open - there is no reason for mixerclient to default to open, and no need for pilot to send redundant configs.

File: pilot/pkg/networking/plugin/mixer/mixer.go [hunk 1]
```diff
@@ -184,14 +185,22 @@ func buildTransport(mesh *meshconfig.MeshConfig, uid attribute) *mccpb.Transport
 		port = mixerMTLSPortNumber
 	}
 
-	return &mccpb.TransportConfig{
+	res := &mccpb.TransportConfig{
 		CheckCluster:  model.BuildSubsetKey(model.TrafficDirectionOutbound, "", model.Hostname(policy), port),
 		ReportCluster: model.BuildSubsetKey(model.TrafficDirectionOutbound, "", model.Hostname(telemetry), port),
+	}
+	// TODO(yangminzhu): remove this after the default on client code is changed.
```
[the remark this answers is not in the document]
Comment (costinm) [reply pilot/pkg/networking/plugin/mixer/mixer.go]: We don't want to maintain fail open - there is no such backward compat constraint. 

Just that new envoy works properly ( fail close ).

**Comment by costinm** [comment]
Lint failure in file not touched by change, probably metalinter issue to be resolved by separate PR.

**Pull request merged**